app/components/go_component.rb

```diff
@@ -0,0 +1,45 @@
+class GoComponent < ViewComponent::Base
+  include Motion::Component
+
+  map_motion :place
+  map_motion :reset
+
+  def initialize(key:)
+    @key = key
+    @game = GoGame.find(key: key) || GoGame.new(key: key)
+    puts "component game #{@game}"
+    @board = @game.display
+    @current = @game.current
+    @captures = @game.captures
+
+    stream_from "go:#{@key}", :player_move
+  end
+
+  def place(event)
+    index = event.target.data[:index].to_i
+    pos = GoGame::Pos.new((index / 9), index.modulo(9))
```

Comment:
My understanding is that this is hard-coding in a game size of 9.

Should we add a comment here, or check the size from the game instead?

Reply:
yeah i'll add a comment. i don't have the assets yet for making the larger size boards. probably there will be a button to determine the size. i was thinking i might want to make a public method on the game for doing this instead of directly building the new position struct here in the component anyway, so yes, the game object can determine the correct position based on the board size.